### ISSUE TITLE
String vs. Symbol vs. InstanceName

### ISSUE THREAD

**Issue by gitttt** [noxdafox/clipspy#16]
The below code outputs:

```
---
/path/to/binary
<class 'str'>
---
/path2/to/binary
<class 'clips.common.Symbol'>
---
/path2/to/binary
<class 'str'>

```

Is this how it is supposed to be? Isn't there a type InstanceName? How can I know from python if I have `(slotname [a_string])`
or

`(slotname "a_string")`

?

```
import clips


env = clips.Environment()


binary_class_string = """
(defclass clips_BINARY (is-a USER)
  (slot path)
  (slot path2)
  (slot path3))
"""
env.build(binary_class_string)

network_function_class_string = """
(defclass clips_NETWORK_FUNCTION (is-a USER)
  (slot binary)
  (slot start_address))
"""
env.build(network_function_class_string)


make_binary_instance_string = """
(make-instance UniversalSample.exe of clips_BINARY (path [/path/to/binary]) (path2 /path2/to/binary) (path3 "/path2/to/binary"))
"""

env.eval(make_binary_instance_string)
instance_just_created = env.find_instance("UniversalSample.exe")
print(instance_just_created)

make_network_function_instance_string = """
(make-instance my_func of clips_NETWORK_FUNCTION (binary [UniversalSample.exe])(start_address 4278884))
"""

env.eval(make_network_function_instance_string)
instance_just_created = env.find_instance("my_func")
print(instance_just_created)
print(type(instance_just_created["binary"]))
linked_instance = env.find_instance(instance_just_created["binary"])
print(linked_instance)

print("---")
print(linked_instance['path'])
print(type(linked_instance['path']))

print("---")
print(linked_instance['path2'])
print(type(linked_instance['path2']))

print("---")
print(linked_instance['path3'])
print(type(linked_instance['path3']))
```

**Comment by noxdafox** [comment]
This is indeed a bug. 

From chapter `2.3.1 Data Types`:

> An instance-name is formed by enclosing a symbol within left and right brackets.

Therefore, `type(instance_just_created['binary'])` should be `Symbol` and not `str`. Yet, it would still remain difficult to distinguish an instance name from a `Symbol` on the Python side. What I could do, is add another class `InstanceName` which inherits from `Symbol`.

Keep in mind that `Symbol` already inherits from `str` so the inheritance chain would be: `str` <-- `Symbol` <-- `InstanceName`.

This means you will need to pay attention if you are discriminating the types as `isinstance(InstanceName('foo'), str)` will return `True` similarly as for `isinstance(True, int)`.

In other words, this is the right comparison order:

```python
if isinstance(var, InstanceName):
    print("%s is an Instance Name" % var)
elif isinstance(var, Symbol):
    print("%s is a Symbol" % var)
elif isinstance(var, str):
    print("%s is a String" % var)
```

**Comment by gitttt** [comment]
Having InstanceName would be the right solution, indeed. It is how pyclips solves this.

**Comment by noxdafox** [comment]
Issue fixed in 985d710a11521975d0a4db98f7daa6597de345af, will make a release next weekend.

**Comment by noxdafox** [comment]
This issue should be fixed in version `0.3.2`. The new release also fixes a crash occurring with user-defined routers.

Windows builds 3.5-3.7 have been uploaded too.
